### PR TITLE
Better setlangcookie param validation

### PR DESCRIPTION
**Pull request by cpeel**
Last night's php_errors report showed some notices after someone was poking around in `tools/setlangcookie.php`. This addresses a few of them by tightening up param validation and enforcing param types in `metarefresh()`. I vetted the param types of every `metarefresh()` call in the code with this change and cleaned up a few optional parameters on some of the calls while I was there.

Sandbox: https://www.pgdp.org/~cpeel/c.branch/better_param_vetting/

### CODE REVIEW

**Review by chrismiceli — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

File: pinc/metarefresh.inc [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * client (determined if headers have been sent or if the output buffer
  * has contents in it).
  */
-function metarefresh($seconds, $url, $title = "", $body = "", $allow_external = false)
+function metarefresh(int $seconds, string $url, string $title = "", string $body = "", bool $allow_external = false)
```
Comment (jchaffraix) [inline pinc/metarefresh.inc:13]: Nit: We may want to add a PHP return as this function should always `exit` for security reason.

```suggestion
/** @return never */
function metarefresh(int $seconds, string $url, string $title = "", string $body = "", bool $allow_external = false)
```

**Review by jchaffraix — APPROVED** [review]
This looks great. The suggestion can be a follow-up if you want to merge this sooner rather than later!

File: pinc/metarefresh.inc (hunk 1 above)
Comment (cpeel) [reply pinc/metarefresh.inc:14]: Great suggestion - thanks!

Comment (cpeel) [reply pinc/metarefresh.inc:14]: Needed to remove some `exit;`s that were after calls to this function.

**Review by jchaffraix — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

**Pull request merged**